### PR TITLE
RFR: Sensu handler improvements and config cleanup

### PR DESCRIPTION
**Pull request by lakshmi-kannan**
## What changed? 

* - [X] Switch to using arparse
* - [X] More detailed error messages
* - [X] A --verbose option 
* - [X] Support --unauthed (st2 server auth disabled) mode.
* - [X] Cleanup config
* - [X] API key support
* - [X] Configurable SSL verify/no verify support.
* - [X] Testing [See https://gist.github.com/lakshmi-kannan/e56feb6a3f2feafb9271]
* - [X] Unit tests
* - [X] Changes in test infrastructure (https://github.com/StackStorm/st2/pull/2505)

## Why?

https://github.com/StackStorm/st2contrib/issues/391
https://github.com/StackStorm/st2contrib/issues/389

## How to test?
See https://gist.github.com/lakshmi-kannan/e56feb6a3f2feafb9271

### CODE REVIEW

**Comment by estee-tew** [comment]
https://stackstorm.slack.com/archives/stackstorm/p1455320630007486

**Comment by manasdk** [comment]
```
{"timestamp":"2016-02-12T15:50:50.569048-0800","level":"info","message":"handler output","handler":{"type":"pipe","command":"/etc/sensu/handlers/st2_handler.py /etc/sensu/handlers/st2_handler.conf","name":"st2"},"output":["Traceback (most recent call last):\n  File \"/etc/sensu/handlers/st2_handler.py\", line 326, in <module>\n    unauthed=args.unauthed)\n  File \"/etc/sensu/handlers/st2_handler.py\", line 311, in main\n    _post_event_to_st2()\nTypeError: _post_event_to_st2() takes at least 1 argument (0 given)\n"]}
```

**Comment by lakshmi-kannan** [comment]
@murgurgula: You might be interested in the changes. It's still WIP.

**Comment by Kami** [comment]
Nice.

Now the only thing it's missing is some tests :trollface:

**Comment by lakshmi-kannan** [comment]
@Kami: yeah, let me try to add some unit tests. This is not going to end well :P.

**Comment by lakshmi-kannan** [comment]
@Kami 7fe2f46 for tests. Took a while. le sigh.

**Comment by lakshmi-kannan** [comment]
Bah. The travis failures are confusing as hell. I think this might have to do with the fact that we just retest packs that changed?

**Comment by lakshmi-kannan** [comment]
@Kami Wdyt about this change? 4178c868043d0e17715bf43b97a6c8beb425d69c
With pylint enabled on test code, I get errors like:
```
************* Module test_sensu_handler
E:126, 8: Function 'post' has no 'assert_called_once_with' member (no-member)
E:137, 8: Function 'post' has no 'assert_called_with' member (no-member)
make: *** [.pylint] Error 123
```
when using mock.  Related: https://lists.logilab.org/pipermail/python-projects/2013-January/003402.html 
I am not sure if there is a work around for this.

**Comment by lakshmi-kannan** [comment]
Seriously, these travis errors are pissing me off!

File: packs/sensu/requirements-tests.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+responses>=0.5.0
```
Comment (Kami) [inline packs/sensu/requirements-tests.txt]: Cool, this is handy.

**Pull request merged**